src/test/java/uk/gov/hmcts/reform/sendletter/services/ReportsServiceTest.java

```diff
@@ -0,0 +1,129 @@
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.sendletter.util.TimeZones.EUROPE_LONDON;
+
+@ExtendWith(MockitoExtension.class)
+class ReportsServiceTest {
+
+    @Mock
+    LettersCountSummaryRepository repository;
+
+    @Mock
+    private ZeroRowFiller zeroRowFiller;
+
+    @Mock
+    ServiceFolderMapping serviceFolderMapping;
+
+    private ReportsService service;
+
+    @BeforeEach
+    void setUp() {
+        this.service = new ReportsService(this.repository, serviceFolderMapping, zeroRowFiller);
+
+        when(this.zeroRowFiller.fill(any()))
+            .thenAnswer(invocation -> invocation.getArgument(0)); // return data unchanged
+    }
+
+    @Test
+    void should_return_letters_count_for_the_date() {
+        LocalDate date = LocalDate.now();
+        LocalTime timeFrom = LocalTime.of(17, 0, 0);
+        LocalTime timeTo = LocalTime.of(16, 0, 0);
+
+        //given
+        given(repository.countByDate(formatDateTime(date.minusDays(1), timeFrom), formatDateTime(date, timeTo)))
+            .willReturn(asList(
+                new ServiceLettersCount("aService", 10),
+                new ServiceLettersCount("bService", 20)
+            ));
+
+        given(serviceFolderMapping.getFolderFor("aService")).willReturn(Optional.of("ServiceA"));
+        given(serviceFolderMapping.getFolderFor("bService")).willReturn(Optional.of("ServiceB"));
+
+        //when
+        List<LettersCountSummary> result = service.getCountFor(date);
+
+        //then
+        assertThat(result)
+            .isNotEmpty()
+            .hasSize(2)
+            .usingFieldByFieldElementComparator()
+            .containsExactlyInAnyOrder(
+                new LettersCountSummary("ServiceA", 10),
+                new LettersCountSummary("ServiceB", 20));
+    }
+
+    @Test
+    void should_return_letters_count_excluding_the_test_service() {
+        LocalDate date = LocalDate.now();
+        LocalTime timeFrom = LocalTime.of(17, 0, 0);
+        LocalTime timeTo = LocalTime.of(16, 0, 0);
+
+        //given
+        given(repository.countByDate(formatDateTime(date.minusDays(1), timeFrom), formatDateTime(date, timeTo)))
+            .willReturn(asList(
+                new ServiceLettersCount("aService", 10),
+                new ServiceLettersCount("send_letter_tests", 20)
+            ));
+        given(serviceFolderMapping.getFolderFor("aService")).willReturn(Optional.of("ServiceA"));
+        given(serviceFolderMapping.getFolderFor("send_letter_tests")).willReturn(Optional.of("BULKPRINT"));
+
+        //when
+        List<LettersCountSummary> result = service.getCountFor(date);
+
+        //then
+        assertThat(result).isNotEmpty()
+            .hasSize(1)
+            .usingFieldByFieldElementComparator()
+            .containsExactlyInAnyOrder(new LettersCountSummary("ServiceA", 10));
+    }
+
+    @Test
+    void should_return_empty_list_for_the_date_when_repo_returns_empty_collection() {
```

Comment:
should it be not we return 0 when we don't have any records for a service ?

Reply:
yeah it returns 0 when no letters uploaded for the service. I'll change the test name

Reply:
@nitinprabhu `ZeroRowFiller` adds rows with zeros, which in this test is mocked to not do anything

Reply:
yeah, I just renamed the test